libcalico-go/lib/errors/errors.go

```diff
@@ -202,3 +203,50 @@ type ErrorParsingDatastoreEntry struct {
 func (e ErrorParsingDatastoreEntry) Error() string {
 	return fmt.Sprintf("failed to parse datastore entry key=%s; value=%s: %v", e.RawKey, e.RawValue, e.Err)
 }
+
+type ErrorPolicyConversion struct {
+	ErrorPolicyConversionRules []ErrorPolicyConversionRule
+}
+
+type ErrorPolicyConversionRule struct {
+	EgressRule  *networkingv1.NetworkPolicyEgressRule
+	IngressRule *networkingv1.NetworkPolicyIngressRule
+	Reason      string
+}
+
+func (e ErrorPolicyConversionRule) String() string {
+	var fieldString string
+
+	switch {
+	case e.EgressRule != nil:
+		fieldString = fmt.Sprintf("%v", e.EgressRule)
```

Comment:
Might want to use "+v" here as your formatter to get field names as well as values.

Reply:
thanks! I knew I was missing something heh